### PR TITLE
Adds eslint

### LINKED ISSUE #53: Add lint support
It is really hard to speculate stringent formatting and coding styles. I recommend you create lint file so that editors will automatically pick it up. Also, it will help external/internal contributors.
See PR #51 
@kencheeto

### PR DESCRIPTION
**Pull request by kencheeto**
Fixes #53. Adds eslint to the project, fixes most errors automatically with --fix and the remainder manually.

I advise setting up an editor integration for eslint - there are many available here: http://eslint.org/docs/user-guide/integrations#editors

* adds eslint to this project
* runs eslint with --fix to automatically fix most errors
* manually fixes several remaining eslint violations
* adds package-lock.json (new with npm 5, replaces yarn.lock)

### CODE REVIEW

**Comment by kencheeto** [comment]
cc @bijujoseph

File: scripts/get-quality-measures-from-pdfs.js [hunk 1]
```diff
@@ -72,19 +72,19 @@ function getRates(path, cb) {
       var remainingFile = data.substring(reportingFound.index + reportingFound[0].length);
       // find the 1, see if a ) or a . comes after each performance rate number
       var firstPos = remainingFile.indexOf('1');
-      var delineationChar = remainingFile[firstPos+1];
+      var delineationChar = remainingFile[firstPos + 1];
 
-      for (var i = 1; i<=numOfRates; i++) {
+      for (var i = 1; i <= numOfRates; i++) {
         // look for the description between 1) and 2)
-        var rateRegex = new RegExp(i + '\\' + delineationChar + '((.|\\s)*?)(' + (i+1)+ '\\' + delineationChar + '|Version 1\\.0)');
```
Comment (samskeller) [inline scripts/get-quality-measures-from-pdfs.js]: Not sure I understand why the `var` was removed here? Cause the only other place it's defined is above if `(!found)` is `true`, but in that case we won't get to this block of code? Or at least should there be a `let rateRegex` before the if-block and no `var` statements used?

File: scripts/get-quality-measures-from-pdfs.js [hunk 2]
```diff
@@ -72,19 +72,19 @@ function getRates(path, cb) {
...
-        if (i+1 > numOfRates) {
+        if (i + 1 > numOfRates) {
           // if this is the last rate, look for the description between x) and the ending phrases
-          var endingPhrases = '(REPORTING CRITERIA|Measure Reporting:|Version 1\\.0|DENOMINATOR \\(REPORTING|The eligible clinician should submit data)'
-          rateRegex = new RegExp(i +'\\' + delineationChar + '((.|\\s)*?)' + endingPhrases);
+          var endingPhrases = '(REPORTING CRITERIA|Measure Reporting:|Version 1\\.0|DENOMINATOR \\(REPORTING|The eligible clinician should submit data)';
+          rateRegex = new RegExp(i + '\\' + delineationChar + '((.|\\s)*?)' + endingPhrases);
         }
 
-        var foundRate = remainingFile.match(rateRegex);
```
Comment (samskeller) [inline scripts/get-quality-measures-from-pdfs.js]: Same question about the `var` removal here

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  extends: 'nava',
+  env: {
+    mocha: true
+  }
+}
```
Comment (samskeller) [inline .eslintrc]: Just for my own learning -- does the linter automatically get run when you do `npm test` if you include this file? Or how does that auto-running of the linter work?

**Comment by bijujoseph** [comment]
Thanks @kencheeto

File: .eslintrc (hunk 3 above)
Comment (kencheeto) [inline .eslintrc]: npm will always run a pretest script if you've defined one before a test script - see package.json (`"pretest": "eslint test util scripts index.js"`)

File: scripts/get-quality-measures-from-pdfs.js (hunk 1 above)
Comment (kencheeto) [inline scripts/get-quality-measures-from-pdfs.js]: you're right - using `let` is better with the intention of the code

**Pull request merged**